content/ytt/docs/latest/lang-ref-ytt-schema.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Schema module
+---
+
+## Overview
+
+`ytt` schemas are currently in the **experimental** phase. To use schema features, include `--enable-experiment-schema`.
+
+`ytt`'s Schema feature provides a way to verify data value YAML structures and values with the help of annotations.
+
+A schema allows you to provide default data values, and verify that each data value file provided is type checked and validated against the schema and guaranteed to be proper.
+
+---
+## Defining Schema
+Configuration Authors establish a Schema by capturing the structure in a YAML document with `#@schema/match` at the top:
+```yaml
+#@schema/match data_values=True
+---
+#! Schema contents
+```
+- `data_values` (`bool`) — whether this schema is applicable to data values. If this
+  is not set to `True`, this results in an error.
+
+Notes:
+- Schema must be defined _before_ any other files containing YAML can be processed. Therefore, a file containing a Schema document must not contain other kinds of documents.
```

Comment:
```suggestion
- a file containing a Schema document must not contain other kinds of documents. This is because schema must be defined _before_ templates can be processed.
```

- The most important bit of information for the user is the rule... and _then_ the rational.
- we can narrow the set of files to just "templates", here. They are the only files that can load data values. and I believe it helps to make it (slightly) more concrete.